modules/library/main/src/main/java/org/geotools/styling/SLDParser.java

```diff
@@ -2019,6 +1997,56 @@ protected Stroke parseStroke(Node root) {
         return stroke;
     }
 
+    private List<Expression> parseDashArray(Node root) {
+        NodeList children = root.getChildNodes();
+        List<Expression> expressions = new ArrayList<>();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child == null) continue;
+            switch (child.getNodeType()) {
+                case Node.TEXT_NODE:
+                    handleDashArrayText(child.getNodeValue(), expressions);
+                    break;
+                case Node.ELEMENT_NODE:
+                    handleDashArrayNode(child, expressions);
+                    break;
+                case Node.CDATA_SECTION_NODE:
+                    handleDashArrayText(child.getNodeValue(), expressions);
+                    break;
+            }
+        }
+        return expressions;
+    }
+
+    private void handleDashArrayText(String text, List<Expression> expressions) {
+        if (text == null || text.isEmpty()) {
+            return;
+        }
+        for (String textPart : text.split("\\s+")) {
```

Comment:
Thank you, this is a more robust and cleaner way to tokenize a string separated than spaces, compared to the original code